radixdlt-core/radixdlt/src/main/java/org/radix/network/messages/PeersResponseMessage.java

```diff
@@ -64,29 +64,28 @@
 
 package org.radix.network.messages;
 
-import java.util.Objects;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.collect.ImmutableSet;
 import com.radixdlt.network.p2p.RadixNodeUri;
-import org.radix.network.messaging.Message;
-
 import com.radixdlt.serialization.DsonOutput;
 import com.radixdlt.serialization.DsonOutput.Output;
 import com.radixdlt.serialization.SerializerId2;
+import org.radix.network.messaging.Message;
 
-import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
 
 @SerializerId2("p2p.discovery.peers_response")
 public final class PeersResponseMessage extends Message {
 	@JsonProperty("peers")
 	@DsonOutput(Output.ALL)
 	private final ImmutableSet<RadixNodeUri> peers;
 
-	public PeersResponseMessage() {
-		this(ImmutableSet.of());
-	}
-
-	public PeersResponseMessage(ImmutableSet<RadixNodeUri> peers) {
-		this.peers = peers;
+	@JsonCreator
+	public PeersResponseMessage(
+		@JsonProperty("peers") ImmutableSet<RadixNodeUri> peers
+	) {
+		this.peers = peers == null ? ImmutableSet.of() : peers;
```

Comment:
Can it be empty?

Reply:
Perhaps not, but, at least, it will not trigger exceptions.

Reply:
Could it trigger exceptions during processing?

Reply:
If there will be `null`, then NPE will be thrown here:
(PeerDiscovery.java)
```
	public RemoteEventProcessor<PeersResponse> peersResponseRemoteEventProcessor() {
		return (sender, peersResponse) -> {
			final var senderNodeId = NodeId.fromPublicKey(sender.getKey());
			if (!peersAsked.contains(senderNodeId)) {
				log.warn("Received unexpected peers response from {}", senderNodeId);
				this.peerControl.banPeer(senderNodeId, Duration.ofMinutes(15), "Unexpected peers response");
				return;
			}

			this.peersAsked.remove(senderNodeId);
			final var peersUpToLimit = peersResponse.getPeers()  ///<<<<<<<<<<<<<<-------- NPE
				.stream()
				.limit(MAX_PEERS_IN_RESPONSE)
				.collect(ImmutableSet.toImmutableSet());
			this.addressBook.addUncheckedPeers(peersUpToLimit);
		};
	}
```